### PR TITLE
feat(error): report error messages using x-error header

### PR DESCRIPTION
**Pull request by trieloff**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -216,6 +216,9 @@ async function executeActions(req, context, params) {
       log.error(`no valid response could be fetched: ${e}`);
       return new Response('', {
         status: /* istanbul ignore next */ e.statusCode === 502 ? 504 : e.statusCode,
+        headers: {
+          'x-error': e.message,
```
Comment (tripodsan) [inline src/index.js:220]: shoulnd't we try to aggregate the errors from the down-stream errors?

Comment (trieloff) [reply src/index.js:220]: `helix-dispatch` is failing fast, so the first error will get propagated, even if there are more than one. I don't think it is worth to wait for the remaining errors, especially when this gets us close to the 30 second limit.

Comment (tripodsan) [reply src/index.js:220]: no, I mean, aggregate the `x-error` headers from the downstream actions. for example, if `static` returns with a `500` and has a `x-error` header, then combine it here.

**Review by tripodsan — APPROVED** [review]

**Pull request merged**